os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -798,6 +789,209 @@ bool libspdm_gen_csr(uint32_t base_hash_algo, uint32_t base_asym_algo, bool *nee
+
+    csr_buffer_size = *csr_len;
+
+    /*device gen csr need reset*/
+    if (*need_reset) {
+        result = libspdm_read_cached_last_csr_request(&cached_last_csr_request,
+                                                      &cached_last_request_len,
+                                                      1, &rsp_csr_tracking_tag);
+
+        /*get the cached last csr request and csr*/
+        if ((result) &&
+            (cached_last_request_len == request_size) &&
+            (libspdm_consttime_is_mem_equal(cached_last_csr_request, request,
+                                            request_size)) &&
+            (libspdm_read_cached_csr(&cached_csr, csr_len)) &&
+            (*csr_len != 0)) {
+
+            /*get and save cached csr*/
+            if (csr_buffer_size < *csr_len) {
+                free(cached_csr);
+                free(cached_last_csr_request);
+                LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO,
+                               "csr buffer is too small to store cached csr! \n"));
+                return false;
+            } else {
+                libspdm_copy_mem(csr_pointer, csr_buffer_size, cached_csr, *csr_len);
+            }
+
+            /*device don't need reset this time*/
+            *need_reset = false;
+
+            free(cached_csr);
+            free(cached_last_csr_request);
+            return true;
+        } else {
+            if (cached_last_csr_request != NULL) {
+                free(cached_last_csr_request);
+            }
+
+            /*device need reset this time: cache the last_csr_request */
+            result = libspdm_cache_last_csr_request(request, request_size, 1);
+            if (!result) {
+                return result;
+            }
+
+            /*device need reset this time*/
+            *need_reset = true;
+            return true;
+        }
+    } else {
+        result = libspdm_gen_csr_without_reset(base_hash_algo, base_asym_algo,
+                                               requester_info, requester_info_length,
+                                               opaque_data, opaque_data_length,
+                                               csr_len, csr_pointer, is_device_cert_model);
+        return result;
+    }
+
+}
+
+bool libspdm_gen_csr_ex(uint32_t base_hash_algo, uint32_t base_asym_algo, bool *need_reset,
+                        const void *request, size_t request_size,
+                        uint8_t *requester_info, size_t requester_info_length,
+                        uint8_t *opaque_data, uint16_t opaque_data_length,
+                        size_t *csr_len, uint8_t *csr_pointer,
+                        bool is_device_cert_model,
+                        uint8_t *csr_tracking_tag,
+                        bool overwrite)
+{
+    bool result;
+    uint8_t *cached_last_csr_request;
+    size_t cached_last_request_len;
+    uint8_t *cached_csr;
+    size_t csr_buffer_size;
+    uint8_t rsp_csr_tracking_tag;
+    uint8_t available_csr_tracking_tag;
```

Comment:
`uint8_t *csr_tracking_tag` at line 871
`uint8_t rsp_csr_tracking_tag;` at line 879
`uint8_t available_csr_tracking_tag;` at line 880
The above three variable names are similar, what is the difference? It is hard to understand.

Reply:
Thanks. I changed the `csr_tracking_tag` to `req_csr_tracking_tag`.